### PR TITLE
parsing file paths from foundry.toml

### PR DESCRIPTION
**Pull request by loocapro**
## Motivation
Related to: https://github.com/foundry-rs/foundry/issues/1200

## Solution
Adding `ignored_file_paths` field and parsing it.

### CODE REVIEW

**Comment by DaniPopes** [comment]
Can you pls rebase to run CI https://github.com/foundry-rs/foundry/pull/7064

File: crates/config/README.md [hunk 1]
```diff
@@ -106,6 +106,7 @@ etherscan_api_key = "YOURETHERSCANAPIKEY"
 # known error codes are: ["unreachable", "unused-return", "unused-param", "unused-var", "code-size", "shadowing", "func-mutability", "license", "pragma-solidity", "virtual-interfaces", "same-varname"]
 # additional warnings can be added using their numeric error code: ["license", 1337]
 ignored_error_codes = ["license", "code-size"]
+ignored_file_paths = ["path_to_ignore"]
```
Comment (mattsse) [inline crates/config/README.md:109]: This should be named more explicitly, like `ignored_warnings_from` or similar

**Review by mattsse — CHANGES_REQUESTED** [review]
only naming nits

File: crates/forge/tests/cli/cmd.rs [hunk 2]
```diff
@@ -666,6 +666,45 @@ contract ATest is DSTest {
     }
 );
 
+// test that `forge build` does not print `(with warnings)` if file path is ignored
+forgetest!(can_compile_without_warnings_ignored_file_paths, |prj, cmd| {
+    let create_config =
+        |ignored_file_paths: Vec<PathBuf>, ignored_error_codes: Vec<SolidityErrorCode>| -> Config {
+            Config { ignored_file_paths, ignored_error_codes, ..Default::default() }
+        };
```
Comment (mattsse) [inline crates/forge/tests/cli/cmd.rs:674]: pedantic nitpick: we don't really need this, should just create `Config {.., Default::default}` where used

File: crates/forge/tests/cli/cmd.rs [hunk 3]
```diff
@@ -666,6 +666,45 @@ contract ATest is DSTest {
...
+    // Build and check for no warnings
+    cmd.args(["build", "--force"]);
+    let out = cmd.stdout_lossy();
+    // expect no warning as path is ignored
+    assert!(out.contains("Compiler run successful!"));
+    assert!(!out.contains("Compiler run successful with warnings:"));
+
+    // Reconfigure without ignored paths or error codes and check for warnings
+    let config_no_ignored = create_config(vec![], vec![]);
+    prj.write_config(config_no_ignored);
```
Comment (mattsse) [inline crates/forge/tests/cli/cmd.rs:700]:
```suggestion
    prj.write_config(Config::default());
```

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**